shared/js/background/before-request.es6.js

```diff
@@ -97,6 +97,17 @@ function handleRequest (requestData) {
     // don't have a tab instance for this tabId or this is a new requestId.
     if (requestData.type === 'main_frame') {
         if (!thisTab || thisTab.requestId !== requestData.requestId) {
+            if(thisTab) {
+                // upon navigation, remove any custom action session rules
+                const customRulesIds = []
```

Comment:
IMO let's move this logic into your `custom-rules-manager.js` module as a function and import+call it here.

Reply:
👍